Code/GraphMol/Wrap/Mol.cpp

```diff
@@ -199,6 +200,101 @@ namespace RDKit {
     return mol.getNumAtoms(onlyExplicit);
   }
 
+  //----------------------------------------------------------------------------
+  // Atom Bookmark Interface
+  void SetAtomBookmark(ROMol *mol, Atom *atom, int mark)
+  {
+    // dangerous as there is not concept of shared pointers here (or internal
```

Comment:
Looking at the source code, it isn't as dire as it seems.  See: https://github.com/bp-kelley/rdkit/commit/0885c9e8530ac46ce4b1d610a1e2d7f058f3928e

I added some preconditions that enforce the atom has to owned by the molecule as this appears to be the case in the code and also how the RWMol operates.  This ensures that if you hold onto an atom, the mol stays around so it is safe.  I think this is safe across the board for the Python API, but might not be if atoms are moved in C++ land, but RWMols are "advanced" anyway so we may not care.

It may be wise to move the preconditions into the Python API and not the C++ API if this is too constraining for C++, however it does neatly solve the ownership issue and does appear how the bookmarks were designed, i.e. bookmarking your OWN atoms.

Note that I had to do this with the Python API for the Bond bookmarks as the parsers fail with the owning mol precondition.  The atom ones are fine, but this makes me think moving it exclusively to python is appropriate. 

I'm a bit at a loss how we could solve this otherwise, but I'll take a look.